pyiceberg/io/pyarrow.py

```diff
@@ -937,7 +937,7 @@ def primitive(self, primitive: pa.DataType) -> PrimitiveType:
             else:
                 raise TypeError(f"Unsupported precision for timestamp type: {primitive.unit}")
 
-            if primitive.tz == "UTC" or primitive.tz == "+00:00":
+            if primitive.tz in ("UTC", "+00:00", "Etc/UTC"):
```

Comment:
In the future, shall we consider add other representations of UTC, such as `"Z"`?

Reply:
Supernit:
```suggestion
            if primitive.tz in {"UTC", "+00:00", "Etc/UTC"}:
```